components/modules/mqtt.c

```diff
@@ -8,6 +8,24 @@
 #include "mqtt_client.h"
 #include <string.h>
 
+#define MQTT_MAX_HOST_LEN           64
+#define MQTT_MAX_CLIENT_LEN         32
+#define MQTT_MAX_USERNAME_LEN       32
+#define MQTT_MAX_PASSWORD_LEN       65
+#define MQTT_MAX_LWT_TOPIC          32
+#define MQTT_MAX_LWT_MSG            128
+
+typedef struct {
+	esp_mqtt_client_config_t config;
+	char host[MQTT_MAX_HOST_LEN];
+	char uri[MQTT_MAX_HOST_LEN];
+	char client_id[MQTT_MAX_CLIENT_LEN];
+	char username[MQTT_MAX_USERNAME_LEN];
+	char password[MQTT_MAX_PASSWORD_LEN];
+	char lwt_topic[MQTT_MAX_LWT_TOPIC];
+	char lwt_msg[MQTT_MAX_LWT_MSG];
+} esp_mqtt_lua_client_config_t;
```

Comment:
The older version of esp-mqtt included character arrays in `esp_mqtt_client_config_t`. The new version includes a `const char *` to a user-allocated string. In order to add compatibility with the new configuration type I defined a new parent struct for the config that allocates the memory that used to be allocated by `esp_mqtt_client_config_t` and pointed the new fields to that memory.

Reply:
Elegant solution :wink: